providers/threeplay.go

```diff
@@ -89,18 +77,43 @@ func (c *ThreePlayProvider) GetProviderJob(id string) (*database.ProviderJob, er
 func (c *ThreePlayProvider) DispatchJob(job *database.Job) error {
 	jobLogger := c.logger.WithFields(log.Fields{"JobID": job.ID, "Provider": job.Provider})
 	query := url.Values{}
-
+	turnaroundLevel := "asr"
 	for k, v := range job.ProviderParams {
-		query.Add(k, v)
+		if k == "turnaround_level_id" {
+			turnaroundLevel = v
+		} else {
+			query.Add(k, v)
+		}
 	}
-	fileID, err := c.UploadFileFromURL(job.MediaURL, query)
+	fileID, err := c.UploadFileFromURL(query)
 
 	if err != nil {
-		jobLogger.Error("Failed to dispatch job to 3Play", err)
+		jobLogger.Error("Failed to upload file to 3Play", err)
 		return err
 	}
 
-	job.ProviderParams["ProviderID"] = strconv.FormatUint(uint64(fileID), 10)
+	transcriptResponse, err := c.OrderTranscript(strconv.Itoa(fileID), "", turnaroundLevel)
+	if err != nil {
+		jobLogger.Error("Failed to order caption", err)
+	}
+
+	job.ProviderParams["ProviderID"] = strconv.Itoa(transcriptResponse.ID)
 
 	return nil
 }
+
+// CancelJob cancels a job if it is in a cancellable state
+func (c *ThreePlayProvider) CancelJob(id string) (bool, error) {
+	providerJob, err := c.GetProviderJob(id)
+	if err != nil {
+		return false, err
+	}
+	if providerJob.Cancellable {
+		err = c.CancelTranscript(providerJob.ID)
+		if err != nil {
+			return providerJob.Cancellable, err
```

Comment:
Shouldn't this be:

```suggestion
			return false, err
```

(if CancelTranscript fails, ``canceled`` should be false, no?)

Reply:
I suppose there's a scenario where the job is "cancellable" on the 3play side but the actual cancellation fails for some reason, that's why I wanted to pass the bool instead of interpreting it.

The way I saw this working is - if 3play says we can cancel the job, try doing it. If that fails, give an error and pass back the cancellable status. Then, it's up to whoever called `CancelJob` to decide whether to try again based on that.

If it's not cancellable, return that bool (I guess we could just say `false` in that last return case if you think that's more clear).

Reply:
Yeah this makes sense. I assumed this meant cancelled, not cancellable x)